tests/libres_tests/res/job_queue/test_job_queue.py

```diff
@@ -311,3 +311,71 @@ async def test_retry_on_closed_connection(tmpdir):
     await job_queue.stop_jobs_async()
     while job_queue.isRunning:
         await asyncio.sleep(0.1)
+
+
+class MockedJob:
+    def __init__(self, status):
+        self.status = status
+        self._start_time = 0
+        self._current_time = 0
+
+    @property
+    def runtime(self):
+        return self._end_time - self._start_time
+
+    def stop(self):
+        self.status = JobStatusType.JOB_QUEUE_FAILED
+
+    def convertToCReference(self, _):
+        pass
+
+
+def test_stop_long_running():
+    """
+    This test should verify that only the jobs that have a runtime
+    25% longer than the average completed are stopped when
+    stop_long_running_jobs is called.
+    """
+    job_list = [MockedJob(JobStatusType.JOB_QUEUE_WAITING) for _ in range(10)]
+
+    for i in range(5):
+        job_list[i].status = JobStatusType.JOB_QUEUE_DONE
+        job_list[i]._start_time = 0
+        job_list[i]._end_time = 10
+
+    for i in range(5, 8):
+        job_list[i].status = JobStatusType.JOB_QUEUE_RUNNING
+        job_list[i]._start_time = 0
+        job_list[i]._end_time = 20
+
+    for i in range(8, 10):
+        job_list[i].status = JobStatusType.JOB_QUEUE_RUNNING
+        job_list[i]._start_time = 0
+        job_list[i]._end_time = 5
+
+    # The driver is of no consequence, so resolving it in the c layer is
+    # uninteresting and mocked out.
+    with patch("res.job_queue.JobQueue._set_driver"):
+        queue = JobQueue(MagicMock())
+
+        # We don't need the c layer call here, we only need it added to
+        # the queue's job_list.
+        with patch("res.job_queue.JobQueue._add_job") as _add_job:
+            for idx, job in enumerate(job_list):
+                _add_job.return_value = idx
+                queue.add_job(job, idx)
+
+    queue.stop_long_running_jobs(5)
+    queue._differ.transition(queue.job_list)
+
+    for i in range(5):
+        assert job_list[i].status == JobStatusType.JOB_QUEUE_DONE
+        assert queue.snapshot()[i] == str(JobStatusType.JOB_QUEUE_DONE)
+
+    for i in range(5, 8):
+        assert job_list[i].status == JobStatusType.JOB_QUEUE_FAILED
+        assert queue.snapshot()[i] == str(JobStatusType.JOB_QUEUE_FAILED)
+
+    for i in range(8, 10):
+        assert job_list[i].status == JobStatusType.JOB_QUEUE_RUNNING
+        assert queue.snapshot()[i] == str(JobStatusType.JOB_QUEUE_RUNNING)
```

Comment:
This might be a better assert: 
```suggestion
    result = [(job.status, item) for (job, item) in zip(job_list, queue.snapshot())]
    assert result == [
        (JobStatusType(name="JOB_QUEUE_DONE", value=32), 0),
        (JobStatusType(name="JOB_QUEUE_DONE", value=32), 1),
        (JobStatusType(name="JOB_QUEUE_DONE", value=32), 2),
        (JobStatusType(name="JOB_QUEUE_DONE", value=32), 3),
        (JobStatusType(name="JOB_QUEUE_DONE", value=32), 4),
        (JobStatusType(name="JOB_QUEUE_FAILED", value=8192), 5),
        (JobStatusType(name="JOB_QUEUE_FAILED", value=8192), 6),
        (JobStatusType(name="JOB_QUEUE_FAILED", value=8192), 7),
        (JobStatusType(name="JOB_QUEUE_RUNNING", value=16), 8),
        (JobStatusType(name="JOB_QUEUE_RUNNING", value=16), 9),
    ]
```
know you are only moving this code, so do it if you agree. The advantage is one assert instead of many, so the feedback on failure is hopefully better.